csharp/ql/test/library-tests/dataflow/arrays/ArrayFlow.cs

```diff
@@ -0,0 +1,229 @@
+// semmle-extractor-options: /r:System.Linq.dll
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+public class A
+{
+    public void M1()
+    {
+        var a = new A();
+        var @as = new[] { a };
+        Sink(@as[0]); // flow
+        SinkElem(@as); // flow
+        Sink(First(@as)); // flow
+    }
+
+    public void M2(A other)
+    {
+        var a = new A();
+        var @as = new[] { other };
+        Sink(@as[0]); // no flow
+        SinkElem(@as); // no flow
+        Sink(First(@as)); // no flow
+    }
+
+    public void M3()
+    {
+        var a = new A();
+        var @as = new A[1];
+        @as[0] = a;
+        Sink(@as[0]); // flow
+        SinkElem(@as); // flow
+        Sink(First(@as)); // flow
+    }
+
+    public void M4(A other)
+    {
+        var a = new A();
+        var @as = new A[1];
+        @as[0] = other;
+        Sink(@as[0]); // no flow
+        SinkElem(@as); // no flow
+        Sink(First(@as)); // no flow
+    }
+
+    public void M5()
+    {
+        var a = new A();
+        var list = new List<A>();
+        list[0] = a;
+        Sink(list[0]); // flow
+        SinkListElem(list); // flow
+        Sink(ListFirst(list)); // flow
+    }
+
+    public void M6(A other)
+    {
+        var list = new List<A>();
+        list[0] = other;
+        Sink(list[0]); // no flow
+        SinkListElem(list); // no flow
+        Sink(ListFirst(list)); // no flow
+    }
+
+    public void M7()
+    {
+        var a = new A();
+        var list = new List<A>() { a };
+        Sink(list[0]); // flow
+        SinkListElem(list); // flow
+        Sink(ListFirst(list)); // flow
+    }
+
+    public void M8(A other)
+    {
+        var list = new List<A>() { other };
+        Sink(list[0]); // no flow
+        SinkListElem(list); // no flow
+        Sink(ListFirst(list)); // no flow
+    }
+
+    public void M9()
+    {
+        var a = new A();
+        var list = new List<A>();
+        list.Add(a);
+        Sink(list[0]); // flow
+        SinkListElem(list); // flow
+        Sink(ListFirst(list)); // flow
+    }
+
+    public void M10(A other)
+    {
+        var list = new List<A>();
+        list.Add(other);
+        Sink(list[0]); // no flow
+        SinkListElem(list); // no flow
+        Sink(ListFirst(list)); // no flow
+    }
+
+    public void M11()
+    {
+        var a = new A();
+        var dict = new Dictionary<int, A>();
```

Comment:
This is a dictionary, not an array. This suggests to me that the test should be moved or renamed so something like "indexed collections".